packages/edit-site/src/components/block-editor/use-site-editor-settings.js

```diff
@@ -128,10 +133,26 @@ export default function useSiteEditorSettings() {
 				select( coreStore ).getBlockPatternCategories(),
 			templateSlug: _record.slug,
 			userPatternCategories: getUserPatternCategories(),
+			settingsInserterMediaCategories:
+				select( blockEditorStore ).getSettings()
+					.inserterMediaCategories || EMPTY_ARRAY,
 		};
 	}, [] );
 	const archiveLabels = useArchiveLabel( templateSlug );
 
+	const mergedInserterMediaCategories = useMemo( () => {
+		const coreInserterMediaCategoriesNames = inserterMediaCategories.map(
+			( { name } ) => name
+		);
+		return [
+			...inserterMediaCategories,
+			...settingsInserterMediaCategories.filter(
+				( { name } ) =>
+					! coreInserterMediaCategoriesNames.includes( name )
+			),
+		];
+	}, [ settingsInserterMediaCategories ] );
```

Comment:
I don't really like this fix personally, basically to pass the correct setting to the block editor, we retrieve the same settings from the block editor. I think that's a bad approach and that would force us to apply the same fix for all block editor instances. 

Instead we should do something similar to what @ellatrix did for the "styles" setting of the block editor setting. Basically have a dedicated reducer in the block editor store that is only about the "registered categories" and "merge" both in the consumer level, not when passing the "settings" prop to the block editor provider.

Reply:
I don't like this fix either TBH. So you mean leave the edit-post and edit-site settings as they are and have a new prop(reducer) in block editor, right?

Reply:
I updated the code. If that's what you had in mind, it feels way more elegant. Thanks!